src/Infiltration/ui/SlashGame.tsx

```diff
@@ -39,24 +40,27 @@ export function SlashGame(props: IMinigameProps): React.ReactElement {
       props.onSuccess();
     }
   }
-  const hasAugment = Player.hasAugmentation(AugmentationName.MightOfAres, true);
-  const guardingTime = Math.random() * 3250 + 1500 - (250 + difficulty.window);
-  const preparingTime = difficulty.window;
-  const attackingTime = 250;
+
+  const guardingTimeRef = useRef(Math.random() * 3250 + 1500 - (250 + difficulty.window));
 
   useEffect(() => {
+    const preparingTime = difficulty.window;
+    const attackingTime = 250;
+
     let id = window.setTimeout(() => {
       setPhase(1);
       id = window.setTimeout(() => {
         setPhase(2);
         id = window.setTimeout(() => props.onFailure(), attackingTime);
       }, preparingTime);
-    }, guardingTime);
+    }, guardingTimeRef.current);
+
     return () => {
       clearInterval(id);
     };
-  }, []);
+  }, [difficulty.window, props]);
```

Comment:
`props` in dependencies look wrong. I need to revisit this place and see if I can use refs to avoid it.

Reply:
The only difference between here and other infiltration files is that you're not destructuring props here. Most of the others have props in the dependency array as well, just in their destructured pieces.